mzmine-community/src/main/java/io/github/mzmine/datamodel/features/types/annotations/MolecularStructureType.java

```
@@ -0,0 +1,113 @@
+import io.github.mzmine.datamodel.features.types.modifiers.GraphicalColumType;
+import io.github.mzmine.datamodel.features.types.modifiers.SubColumnsFactory;
+import io.github.mzmine.datamodel.structures.MolecularStructure;
+import io.github.mzmine.javafx.concurrent.threading.FxThread;
+import io.github.mzmine.modules.visualization.molstructure.MolStructureViewer;
+import io.github.mzmine.modules.visualization.molstructure.Structure2DComponent;
+import io.github.mzmine.modules.visualization.molstructure.StructureTableCell;
+import java.util.List;
+import java.util.logging.Logger;
+import javafx.beans.property.Property;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.scene.Node;
+import javafx.scene.control.TreeTableColumn;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class MolecularStructureType extends DataType<MolecularStructure> implements
+    GraphicalColumType<MolecularStructure> {
+
+  private static final Logger logger = Logger.getLogger(MolecularStructureType.class.getName());
+
+  @Override
+  public @NotNull String getUniqueID() {
+    return "structure_2d";
+  }
+
+  @Override
+  public @NotNull String getHeaderString() {
+    return "Structure";
+  }
+
+  @Override
+  public Property<MolecularStructure> createProperty() {
+    return new SimpleObjectProperty<>();
+  }
+
+  @Override
+  public Class<MolecularStructure> getValueClass() {
+    return MolecularStructure.class;
+  }
+
+  @Override
+  public double getColumnWidth() {
+    return DEFAULT_GRAPHICAL_CELL_WIDTH;
+  }
+
+  @Override
+  public @Nullable Node createCellContent(@NotNull final ModularFeatureListRow row,
+      final MolecularStructure cellData, @Nullable final RawDataFile raw,
+      final AtomicDouble progress) {
+    if (cellData == null) {
+      return null;
+    }
+// not really used as we are setting a different cell factory
+    return new Structure2DComponent(cellData.structure());
```

if this is not supposed to happen we should throw an exception, or?

---

True

---

Throwing exception now explaining what to do instead